File: abc277/d/main.py
n,m=map(int,input().split())
a = sorted(list(map(int,input().split())))

ans,sum_,cnt=[],a[0],a[0]
for i in range(1,n):
  if a[i] == cnt or cnt + 1 == a[i]:
    sum_ += a[i]
    cnt = a[i]
  else:
    ans.append(sum_)
    sum_ = a[i]
    cnt = a[i]

if a[-1] == a[0] or (a[-1] == m-1 and a[0] == 0):
  if len(ans) == 0:
    ans.append(sum_)
  else:
    ans[0] += sum_

print(sum(a)- max(ans))

File: abc277/d/main.cpp
#include <iostream>
#include <map>
#include <set>
#include <queue>
#include <cmath>
#include <vector>
#include <string>
#include <algorithm>
#include <functional>
using namespace std;
#define rep(i,a) for(int i=0;i<a;i++)
#define rrep(i,a,b) for(int i=a;i<=b;i++)
#define all(x) x.begin(), x.end()
typedef long long ll;
ll n,m;
ll sum = 0,ans = 0;
vector<int> a;

int main() {
  cin >> n >> m;

  rep(i,n) {
    int x; cin >> x;
    a.push_back(x);
  }

  rep(i,n) sum += a[i];
  sort(all(a));
  rep(i,n) {
    a.push_back(a[i]+m);
  }

  bool f = true;
  rep(i,n) {
    if(a[i+1] - a[i] >= 2) f = false;
  }
  if(f == true) {
    cout << "0" << endl;
    return 0;
  }

  int cx = 0; ll ret = 0;
  while(cx < 2*n) {
    ret += a[cx]%m;
    ans = max(ans,ret);
    if(a[cx+1]-a[cx] >= 2) {
      ret = 0;
    }
    cx++;
  }
  cout << sum - ans << endl;
  return 0;
}